src/Sarif.Driver/Sdk/AnalyzeOptionsBase.cs

```diff
@@ -94,8 +94,18 @@ public abstract class AnalyzeOptionsBase
         public bool RichReturnCode { get; set; }
 
         [Option(
-            "persist-file-contents",
-            HelpText = "Persist a base64-encoded representation of all referenced files to the log.")]
-        public bool PersistFileContents { get; set; }
+            't', // persist 't'ext
+            "persist-text-contents",
+            HelpText = "Persist a base64-encoded representation of all referenced textual files to the log.")]
```

Comment:
It's not Base64-encoded. It's UTF-8 transcoded.

Reply:
@michaelcfanning It _is_ true. Regardless of the encoding of the original file, the property `fileContent.text` is UTF-8. See §3.2.2, fileContent.text property:
> Since SARIF log files are encoded in UTF-8 ([RFC3629]; see §3.1), this means that if the external file is a text file in any encoding other than UTF-8, the SARIF producer SHALL transcode the text to UTF-8 before assigning it to the text property.

Reply:
I don't think you understand what I'm saying. the --persist-text-contents argument does NOT populate the Contents.Text property (which certainly is UTF-8, by spec). The reason is that we don't have a reliable mechanism to determine the encoding of a file on disk (and what this rewrite does is inject file contents from disk into the SARIF). Because we don't know the original encoding of the file, we simply inject it into Contents.Binary. We know this binary representation refers to a textual file only because of its extension. Make sense?

---
In reply to: [201072947](https://github.com/Microsoft/sarif-sdk/pull/936#discussion_r201072947) [](ancestors = 201072947)

Reply:
Oh my gosh. Then I completely misread the code and the distinction between `--persist-text-contents` and `--persist-binary-contents`. I need to look at the product code again. Back with you shortly.

Reply:
I see now. The comments are correct. `--persist-text-contents` means "Populate `file.content`, but only for text files" and `--persist-binary-content` means "Populate `file.content`, but only for binary files". But you always, only, populate `file.content.binary`.

Reply:
That's it!

---
In reply to: [201077447](https://github.com/Microsoft/sarif-sdk/pull/936#discussion_r201077447) [](ancestors = 201077447)